src/poetry/console/commands/run.py

```diff
@@ -44,7 +45,10 @@ def _module(self) -> Module:
 
         return module
 
-    def run_script(self, script: str | dict[str, str], args: str) -> int:
+    def run_script(self, script: str | dict[str, str], args: list[str]) -> int:
+        env = EnvManager(self.poetry).get()
+        args[0] = f"{env.path}/bin/{args[0]}"
```

Comment:
I am 90% sure that the path will be different for windows. IIRC Windows uses `Scripts` directory in venvs (please verify it).

Reply:
This looks like we should add a test that fails if the path is wrong. (In general, we should always try to add a test for a bugfix.)

Further, it might be better to use `Env.script_dirs` than building the path manually. However, I'm not sure which directory to choose if there is more than one.

Reply:
I will take a look in to `env.script_dirs` and also look to add tests to this PR thanks for the feedback 😅

Reply:
@radoering it looks as if changing to `env.script_dirs[0]` works on Linux but not on windows so I am going to leave the paths as `{env.path}\Scripts\{args[0]}` for Windows and `{env.path}/bin/{args[0]}` for Linux and MacOS as we know this is where the script will exist defined in `tool.poetry.scripts`

Reply:
What's the issue with `script_dirs` on Windows?

What I'm actually wondering is if we have to check which folder is the correct one if there are multiple entries in `script_dirs` or can we be sure that it's `script_dirs[0]` (or maybe `script_dirs[-1]`)? It might be relevant if `env` is a `VirtualEnv` or a `SystemEnv`.

Reply:
According to https://github.com/python-poetry/poetry/blob/ba97fea031aecd5fd31878e15ece24b3ddc55752/src/poetry/masonry/builders/editable.py#L164-L172 scripts are written to the first writable folder in `script_dirs`. Thus, we really should use `script_dirs` and have to check which is the correct folder.